test/generators/shop_model_generator_test.rb

```diff
@@ -29,6 +33,13 @@ class ShopModelGeneratorTest < Rails::Generators::TestCase
     end
   end
 
+  test "injects into application controller" do
+    run_generator
+    assert_file "app/controllers/application_controller.rb" do |controller|
+      assert_match "def shop", controller
+    end
```

Comment:
The engine provides a simple shop model - it will have created a migration for you and setup the SessionRepository to use your shop model. The SessionRepository is really just a connector object from the AuthController to whichever object stores your session (the shop model)

Simple answer - if you ran the recommended generator then any controller that inherits from `ShopifyApp::Authenticated` controller can call `shop` which will return the current shop